### PR TITLE
Add shape inference function of Upsample operator.

### PR DESCRIPTION
**Pull request by rockindy**
Add shape inference function of Upsample operator.

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -1085,7 +1085,21 @@ ONNX_OPERATOR_SET_SCHEMA(
             "T",
             OpSchema::all_tensor_types(),
             "Constrain input and output types to all tensor types.")
-        .SetDoc(Upsample_ver7_doc));
+        .SetDoc(Upsample_ver7_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateShapeAndTypeFromFirstInput(ctx);
+          auto& input_shape = getInputShape(ctx, 0);
```
Comment (houseroad) [inline onnx/defs/tensor/defs.cc]: Please call `hasNInputShapes(ctx, 1)` to make sure we have the shape information of the first input.

**Comment by houseroad** [comment]
@zrphercule this PR won't touch the doc. :-)

File: onnx/defs/tensor/defs.cc [hunk 2]
```diff
@@ -1085,7 +1085,24 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+        .SetDoc(Upsample_ver7_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 1)) {
+            return;
+          }
+          propagateShapeAndTypeFromFirstInput(ctx);
+          auto& input_shape = getInputShape(ctx, 0);
+          auto* output_shape = getOutputShape(ctx, 0);
+          auto* scale_attr = ctx.getAttribute("scales");
+          if (input_shape.dim_size() != scale_attr->floats_size())
```
Comment (bddppq) [inline onnx/defs/tensor/defs.cc]: Could you add `{}` around the body of if?

File: onnx/defs/tensor/defs.cc [hunk 3]
```diff
@@ -1085,7 +1085,24 @@ ONNX_OPERATOR_SET_SCHEMA(
             "T",
             OpSchema::all_tensor_types(),
             "Constrain input and output types to all tensor types.")
-        .SetDoc(Upsample_ver7_doc));
+        .SetDoc(Upsample_ver7_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 1)) {
+            return;
+          }
+          propagateShapeAndTypeFromFirstInput(ctx);
```
Comment (bddppq) [inline onnx/defs/tensor/defs.cc]: You probably only want to copy the input type to output here?

File: onnx/defs/tensor/defs.cc (hunk 2 above)
Comment (rockindy) [reply onnx/defs/tensor/defs.cc]: Sure.

File: onnx/defs/tensor/defs.cc (hunk 3 above)
Comment (rockindy) [reply onnx/defs/tensor/defs.cc]: My original intention is to prepare an output object for modifying. I will use propagateElemTypeFromInputToOutput() for less code instructions.
Thanks for your advice.

**Pull request merged**